components/SubgraphEditor/SubgraphConfig/SelectedContract/SelectedContract.tsx

```diff
@@ -124,6 +124,23 @@ export const SelectedContract = (props: SelectedContractProps) => {
     setJumpToLine,
   } = props
 
+  useEffect(() => {
+    if (mappingFunctionNames.length > 0) {
+      const missingFns = events.reduce(
+        (acc: number[], e, i) => (!mappingFunctionNames.includes(e.handler) ? [...acc, i] : acc),
+        []
+      )
+
+      if (missingFns.length > 0) {
+        updateContract({
+          events: events.map((p, i) =>
```

Comment:
What's happening here? Is it removing the event handler if the function is missing?

If so, we don't want to do that, as a function might get removed, then later added back.

Reply:
Yes, exactly. This has already happened before because if they delete a custom-made fn(not the generated one), when the piece of code from below tries to assign the value to the dropdown, it does not find it, and then it shows an  empty value. The code I added handles the case when it's a generated fn and **it also updates in the storage.**

It only removes the handler though, so we could potentially add an error message saying `The function assigned before has been removed, please add a new handler`.

<img width="779" alt="image" src="https://user-images.githubusercontent.com/12175854/179912371-9a7a67d8-4a87-4c19-85c9-bdd3d02f8f85.png">

Reply:
Hmm, I think displaying an empty value in the dropdown might be preferable, as it still shows the event as "unassigned", but doesn't automatically change the stored state.

The scenario in mind is a user building a subgraph, renaming one of their mapping functions by accident, then undoing the renaming. With this change, if they click to the "config" tab after renaming, then it will remove the link from the event to that function.

Probably not the biggest issue either way, as it's sort-of an edge case.